index.js

```diff
@@ -18,20 +18,41 @@ module.exports = function (cmd, args, opts) {
 				val = stripEof(val);
 			}
 
+			if (parsed.options.encoding !== undefined) {
+				val = new Buffer(val, parsed.options.encoding);
+			}
+
 			return val;
 		};
 
-		var spawned = childProcess.execFile(parsed.command, parsed.args, parsed.options, function (err, stdout, stderr) {
-			if (err) {
-				err.stdout = stdout;
-				err.stderr = stderr;
-				reject(err);
-				return;
-			}
+		var spawned = childProcess.spawn(parsed.command, parsed.args, parsed.options);
```

Comment:
You might as well use the https://github.com/IndigoUnited/node-cross-spawn-async/blob/master/index.js API directly now that we're just spawning.